### PR TITLE
Convert test properties to YAML

### PR DESCRIPTION
**Pull request by mpanne**
And add the needed test config and util functions for it.

### CODE REVIEW

File: src/test/kotlin/de/bund/digitalservice/useid/config/TestApplicationProperties.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import de.bund.digitalservice.useid.util.YamlPropertySourceFactory
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.boot.context.properties.EnableConfigurationProperties
+import org.springframework.context.annotation.PropertySource
+
+@EnableConfigurationProperties(TestApplicationProperties::class)
+@PropertySource("classpath:application.yaml", factory = YamlPropertySourceFactory::class)
+@ConfigurationProperties(prefix = "application")
+class TestApplicationProperties {
+    var staging: StagingProperties? = null
```
Comment (pgurusinga) [inline src/test/kotlin/de/bund/digitalservice/useid/config/TestApplicationProperties.kt:12]: why do we set the value as null? is it possible with `lateinit` as well?

**Review by pgurusinga — COMMENTED** [review]
just a side note:
what do you think if we move all files to e.g `/YamlTestConfigSetup` package, since we won't touch all of these changes (A one-time code-config setup to enable Yaml).

File: src/test/kotlin/de/bund/digitalservice/useid/config/TestApplicationProperties.kt (hunk 1 above)
Comment (mpanne) [reply src/test/kotlin/de/bund/digitalservice/useid/config/TestApplicationProperties.kt:12]: Somehow that leads to errors on initialisation. 
 
Screenshot 2022-08-15 at 10 13 41

Comment (mpanne) [reply src/test/kotlin/de/bund/digitalservice/useid/config/TestApplicationProperties.kt:12]: About your package idea: 
The YAML factory could be part of a "one-time" package which is not touched anymore. The TestApplicationProperties can be reused for any config that should be available in any tests so I would like to keep that in the config package.

Comment (pgurusinga) [reply src/test/kotlin/de/bund/digitalservice/useid/config/TestApplicationProperties.kt:12]: that would be great if we can separate that and also for the TestApplicationProperties, maybe we could add an information about the purpose of the file?

**Review by pgurusinga — APPROVED** [review]

File: src/test/kotlin/de/bund/digitalservice/useid/config/TestApplicationProperties.kt (hunk 1 above)
Comment (mpanne) [reply src/test/kotlin/de/bund/digitalservice/useid/config/TestApplicationProperties.kt:15]: Like that?

**Review by hannesschaletzky — APPROVED** [review]

**Pull request merged**